File: 실버 1 (1문제)/1149_RGB 거리.c
// 참고 링크 : https://moon-gd.tistory.com/71
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#define MAX 1001

int red_array[MAX];
int green_array[MAX];
int blue_array[MAX];

int size = 0, i;
int red, green, blue;

int find_min(int a, int b, int c);

int main() {
	scanf("%d", &size);

	for (i = 0; i < size; i++) {
		scanf("%d %d %d", &red, &green, &blue);

		red_array[i + 1] = green_array[i] < blue_array[i] ? green_array[i] + red : blue_array[i] + red;
		green_array[i + 1] = red_array[i] < blue_array[i] ? red_array[i] + green : blue_array[i] + green;
		blue_array[i + 1] = red_array[i] < green_array[i] ? red_array[i] + blue : green_array[i] + blue;
	}

	printf("%d", find_min(red_array[i], green_array[i], blue_array[i]));

	return 0;
}

int find_min(int a, int b, int c) {
	if (a > b) {
		if (b > c) {
			return c;
		}
		else {
			return b;
		}
	}
	// b >= a
	else {
		if (a > c) {
			return c;
		}
		else {
			return a;
		}
	}
}

File: 실버 1 (3문제)/1149_RGB 거리.py
# 참고 링크 : https://moon-gd.tistory.com/71
import sys
q_input = sys.stdin.readline  # 빠른 입력을 받기 위해서

size = int(q_input())

red_array = [0]
green_array = [0]
blue_array = [0]

for i in range(size):
    red, green, blue = map(int, q_input().split())

    # red 배열 갱신
    if green_array[i] < blue_array[i]:
        red_array.append(red + green_array[i])
    else:
        red_array.append(red + blue_array[i])

    # green 배열 갱신
    if red_array[i] < blue_array[i]:
        green_array.append(green + red_array[i])
    else:
        green_array.append(green + blue_array[i])

    # blue 배열 갱신
    if red_array[i] < green_array[i]:
        blue_array.append(blue + red_array[i])
    else:
        blue_array.append(blue + green_array[i])

print(min(red_array[-1], green_array[-1], blue_array[-1]))